src/averageAges.js

```diff
@@ -55,10 +77,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(mother => people.some(person =>
+    person.mother === mother.name));
+
+  let kids = people.filter(mummy => mummy.mother != null);
+
+  if (onlyWithSon === true) {
+    kids = people.filter(son => son.mother != null && son.sex === 'm');
```

Comment:
The 'onlyWithSon' check can be simplified: `kids = people.filter(person => person.mother && person.sex === 'm');` This way, you filter directly by the conditions for kids, without the need to check for mothers separately.